File: catkin_ws/src/.old_archive/learning_topic/src/pose_subscriber.cpp
// subscribe the topic: /turtle1/pose
// message type: turtlesim::Pose

#include <ros/ros.h>
#include <turtlesim/Pose.h>

void poseCallback(const turtlesim::Pose::ConstPtr& msg){
	// print message received
	ROS_INFO("Turtle pose: x:%0.6f, y:%0.6f",msg->x, msg->y);
}

int main (int argc, char **argv)
{
	// node init. Node name: velocity_publisher
	ros::init(argc, argv, "pose_subscriber");
	
	// create node handle
	ros::NodeHandle n;
	
	// Create subscriber. 
	// Subscribed Topic name: /turtle1/pose
	// register call back function
	ros::Subscriber pose_sub = n.subscribe("/turtle1/pose",10,poseCallback);
	
	// loop, waiting for call back function. If message coming in, call poseCallback
	ros::spin();
	
	return 0;
	
}

File: catkin_ws/src/.old_archive/learning_topic/script/pose_subscriber.py
# !/usr/bin/env python3
# -*- coding: utf-8 -*-
# subscribe the topic /turtle1/pose
# msg type: turtlesim:Pose

import rospy
from turtlesim.msg import Pose

def poseCallback(msg):
	rospy.loginfo("Turtle pose: x:%0.6f, y:%0.6f",msg.x, msg.y)
	
def pose_subscriber():
	# node init
	rospy.init_node('pose_subscriber',anonymous=True)
	
	# Create subscriber. 
	# Subscribed Topic name: /turtle1/pose
	# register call back function
	rospy.Subscriber('/turtle1/pose',Pose, poseCallback)
	
	# set loop freq
	rospy.spin()
			
if __name__=='__main__':
	pose_subscriber()
